File: Leetcode_3151.java
class Solution {
    public boolean isArraySpecial(List<Integer> nums) {
        boolean result = true;
        for (int i = 0; i < nums.size() - 1; i++) {
            if (nums.get(i) % 2 == nums.get(i + 1) % 2) {
                result = false;
                break;
            }
        }
        return result;
    }
}

File: Leetcode_3151.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        result = True
        for i in range(len(nums) - 1):
            if (nums[i] % 2 == nums[i + 1] % 2):
                result = False
                break
        return result
